homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -179,11 +179,15 @@ async def async_step_ssdp(self, discovery_info):
             discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME].split("(", 1)[0].strip()
         )
 
+        mac = discovery_info[ssdp.ATTR_UPNP_SERIAL].upper()
         # Synology NAS can broadcast on multiple IP addresses, since they can be connected to multiple ethernets.
         # The serial of the NAS is actually its MAC address.
-        if self._mac_already_configured(discovery_info[ssdp.ATTR_UPNP_SERIAL].upper()):
+        if self._mac_already_configured(mac):
             return self.async_abort(reason="already_configured")
 
+        await self.async_set_unique_id(mac)
```

Comment:
Nick is right, the other call to set_unqiue_id is with serial.

Reply:
we can merge this now I guess, but wouldn't it make sense to always just use mac ?

Reply:
It is still working as expected because we check for mac in existing entries.

Reply:
Yep, indeed the SSDP serial is the MAC address (without separators) of the NAS, and I override it to the serial when we succeed to login.

Don’t think it should be nice to use the MAC as unique ID permanently because most of the NAS has multiple Ethernet port, so multiple MAC.